### ISSUE TITLE
"no-member" checks is not aware about the scope

### ISSUE THREAD

**Issue by duboviy** [PyCQA/pylint#3880]
### Steps to reproduce
1. Create a file with the following source:
```python
"""Example of pylint scope error"""
from unittest.mock import MagicMock

def test1():
    """No complains here"""
    bigquery = MagicMock()
    def get_table(ds_table):
        return ds_table

    bigquery.get_table = get_table

def test2():
    """Errors here"""
    bigquery = MagicMock()
    destination_table = bigquery.get_table.return_value
    destination_table.num_rows = 0
```
2. Run pylint over the mentioned python source code.

### Current behavior
```
pylint pylint_check.py 
************* Module pylint_check
pylint_check.py:15:24: E1101: Function 'get_table' has no 'return_value' member (no-member)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 5.00/10, +0.00)
```

### Expected behavior
Expected to have no errors, as is second function bigquery is different MagicMock().

### pylint --version output
pylint 2.3.1
astroid 2.2.5
Python 3.7.0 (default, Sep 25 2018, 19:06:08)
[GCC 5.4.0 20160609]

**Comment by cread** [comment]
We have just hit this today ourselves, and this also appears to be a duplicate of #2821

**Comment by Pierre-Sassoulas** [comment]
Thank you for noticing @cread, I'm going to close as duplicate.
